cpp/src/arrow/compute/api_vector.h

```diff
@@ -95,6 +95,13 @@ enum class NullPlacement {
   AtEnd,
 };
 
+enum class Tiebreaker {
```

Comment:
I assume this is specific to the rank function, so perhaps put this inside `RankOptions` (and then perhaps make this a simple `enum` to be able to write e.g. `RankOptions::Lowest`).